website/sales/api/v2/views.py

```diff
@@ -0,0 +1,63 @@
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework import filters
+from rest_framework.generics import (
+    ListAPIView,
+    RetrieveAPIView,
+    get_object_or_404,
+    RetrieveUpdateDestroyAPIView,
+    ListCreateAPIView,
+)
+from rest_framework.permissions import DjangoModelPermissionsOrAnonReadOnly, \
+    IsAuthenticated, DjangoModelPermissions
+
+from sales.api.v2.serializers.order import OrderSerializer, OrderListSerializer
+from sales.api.v2.serializers.shift import ShiftSerializer
+from sales.models.order import Order
+from sales.models.shift import Shift
+
+
+class ShiftListView(ListAPIView):
+    """Returns an overview of all sales shifts."""
+
+    serializer_class = ShiftSerializer
+    queryset = Shift.objects.all()
+    filter_backends = (
+        filters.OrderingFilter,
+        filters.SearchFilter,
+    )
+    ordering_fields = ("start_date", "end_date")
+    permission_classes = [IsAuthenticated]
+
+
+class ShiftDetailView(RetrieveAPIView):
+    serializer_class = ShiftSerializer
+    queryset = Shift.objects.all()
+    permission_classes = [IsAuthenticated]
+
+class OrderListView(ListCreateAPIView):
+    serializer_class = OrderListSerializer
+    permission_classes = [
+        IsAuthenticatedOrTokenHasScope,
+        DjangoModelPermissions,
+    ]
+    required_scopes = ["sales:write"]
+
+    def get_queryset(self):
+        shift = get_object_or_404(Shift, pk=self.kwargs.get("pk"))
+        return Order.objects.filter(shift=shift.pk)
```

Comment:
This does not seem to be the right spot to check if the shift exists to me. You are getting it for the context too and only using it's pk.

Maybe you should get it here, set it to`self.shift` and reuse it in get_serializer_context